src/module.cpp

```diff
@@ -1041,8 +1041,8 @@ module::fuse(const module& m,
     if(map_ins == nullptr)
         map_ins = &default_map_ins;
     insert_params(*this, inputs, *map_ins);
-    auto param_map = m.get_ins_param_map(inputs);
-    for(auto&& [input, param] : param_map)
+    auto param_map = m.get_ins_param_map(inputs, true);
+    for(auto&& [param, input] : param_map)
```

Comment:
Why is this flipped?

Reply:
There are cases where same input instruction is mapped to multiple parameters. 
e.g. `split_fused_reduce(x, y, x)`  

In those cases, having mapping from `input--> param` would de-duplicate it and only add single parameter. 

Later 
https://github.com/ROCm/AMDMIGraphX/blob/ae2b02650767d53b9526f660f1b2017ec4bf56e0/src/module.cpp#L238
Here it won't find parameter in the `map_ins` and would try to add it and fails